var/spack/repos/builtin/packages/wrf/package.py

```diff
@@ -293,6 +292,90 @@ def patch(self):
         filter_file("^#!/bin/csh -f", "#!/usr/bin/env csh", *files)
         filter_file("^#!/bin/csh", "#!/usr/bin/env csh", *files)
 
+    @run_before("configure", when="%aocc@4:")
+    def create_aocc_config(self):
+        param = dict()
+
+        param["MPICC"] = self.spec["mpi"].mpicc
+        param["MPIFC"] = self.spec["mpi"].mpifc
+
+        if self.spec.satisfies("@:4.2.2"):
+            param["CTSM_SUBST"] = "-DWRF_USE_CLM"
+        else:
+            param["CTSM_SUBST"] = "CONFIGURE_D_CTSM"
+
+        if self.spec.satisfies("@4.4.0:"):
+            param["NETCDFPAR_BUILD"] = "CONFIGURE_NETCDFPAR_BUILD"
+        else:
+            param["NETCDFPAR_BUILD"] = ""
```

Comment:
I think you can use a dict literal here. Also, maybe it would be good if the template is stored into its own file.

Reply:
Happy to try and put this template in a separate file.  What is the recommended mechanism for getting access to it from the recipe at build time?